### PR TITLE
[NX-9] Include CORS handling in Spree Starter

### PR DESCRIPTION
**Pull request by pmilewski**

### CODE REVIEW

File: config/initializers/cors.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Avoid CORS issues when API is called from the frontend app.
+# Handle Cross-Origin Resource Sharing (CORS) in order to accept cross-origin AJAX requests.
+
+# Read more: https://github.com/cyu/rack-cors
+
+Rails.application.config.middleware.insert_before 0, Rack::Cors, debug: true, logger: (-> { Rails.logger }) do
```
Comment (damianlegawiec) [inline config/initializers/cors.rb:6]:
```suggestion
Rails.application.config.middleware.insert_before 0, Rack::Cors, debug: ENV.fetch('RACK_CORS_DEBUG', false), logger: (-> { Rails.logger }) do
```

File: config/initializers/cors.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  allow do
+    origins ENV.fetch('ALLOWED_ORIGIN_HOSTS', 'localhost').split(',')
+
+    resource '/cors',
```
Comment (damianlegawiec) [inline config/initializers/cors.rb:10]: @pmilewski why do we need this?

Comment (tniezg) [reply config/initializers/cors.rb:10]: We don't need this.

File: config/initializers/cors.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# Read more: https://github.com/cyu/rack-cors
+
+Rails.application.config.middleware.insert_before 0, Rack::Cors, debug: ENV.fetch('RACK_CORS_DEBUG', false), logger: (-> { Rails.logger }) do
+  allow do
+    origins ENV.fetch('ALLOWED_ORIGIN_HOSTS', 'localhost').split(',')
+
+    resource '*',
+      :headers => :any,
+      :methods => [:get, :post, :delete, :put, :patch, :options, :head],
+      :max_age => 0
```
Comment (damianlegawiec) [inline config/initializers/cors.rb:13]:
```suggestion
      headers: :any,
      methods: %i[get post delete put patch options head],
      max_age: 0
```

**Pull request merged**